hack/externalnode/install-vm.sh

```diff
@@ -141,17 +153,236 @@ EOF
+        docker_uninstall_packages="docker docker-engine docker.io containerd runc"
+        apt-get remove ${docker_uninstall_packages}
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        dependent_packages="ca-certificates curl gnupg lsb-release"
+        cmd="apt-get install -y ${dependent_packages}"
+        execute_command "$cmd" 1m
+        docker_gpg_key="/usr/share/keyrings/docker-archive-keyring.gpg"
+        rm -rf ${docker_gpg_key}
+        docker_download_url="https://download.docker.com/linux/ubuntu"
+        timeout --preserve-status --foreground 10s curl -fsSL ${docker_download_url}/gpg | sudo gpg --dearmor -o ${docker_gpg_key}
+        # Setup a stable repository
+        echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=${docker_gpg_key}] ${docker_download_url} \
+        $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        docker_install_packages="docker-ce docker-ce-cli containerd.io docker-compose-plugin"
+        cmd="apt-get install -y ${docker_install_packages}"
+        execute_command "$cmd" 2m
+        ;;
+        rhel)
+        # Uninstall older versions of Docker.
+        docker_uninstall_packages="docker docker-client docker-client-latest docker-common docker-latest docker-latest-logrotate docker-logrotate docker-engine podman runc"
+        sudo yum remove ${docker_unistall_packages}
+        cmd="sudo yum install -y yum-utils"
+        execute_command "$cmd" 1m
+        # We are using CentOs Docker images for Rhel as suggested in official website.
+        docker_download_url="https://download.docker.com/linux/centos/docker-ce.repo"
+        cmd="sudo yum-config-manager --add-repo ${docker_download_url}"
+        execute_command "$cmd" 1m
+        docker_install_packages="docker-ce docker-ce-cli containerd.io docker-compose-plugin"
+        cmd="sudo yum install -y ${docker_install_packages}"
+        execute_command "$cmd" 1m
+        cmd="sudo systemctl start docker"
+        execute_command "$cmd" 1m
+        ;;
+    esac
+}
+
+# Install all pre-requisite packages needed for installation.
+install_prerequisite_packages() {
+    set +eo pipefail
+    # Install Docker, docker-compose only when running services in containers.
+    docker version
+    if [[ "$?" -ne 0 ]]; then
+        install_container_runtime
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with OVS and antrea-agent container specific parameters.
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  antrea-ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    privileged: true
```

Comment:
why is this changed from specific caps to privileged? My comment only meant cap is not needed when antrea-agent container's privileged is already true, which is redundant, didn't mean every container should be privileged. 
Please adjust it following principle of least privilege.

Reply:
Can you tell if this privilege is fine because different types of process access this OVS container. Should we add other types os privilege as well.

cap_add:
        - NET_ADMIN
        - SYS_NICE
        - SYS_ADMIN
        - IPC_LOCK

Please tell @tnqn

Reply:
I think it should same as the caps in antrea-agent pod manifest